CMakeLists.txt

```diff
@@ -90,6 +90,20 @@ if(CERES_FOUND)
   include_directories(${CERES_INCLUDE_DIRS})
 endif(CERES_FOUND)
 
+find_package(HDF5 1.8 COMPONENTS CXX)
+
+if(HDF5_FOUND)
+  message("-- HDF5 found for checkpointing")
+  message("   Version " + ${HDF5_VERSION})
+  include_directories(${HDF5_INCLUDE_DIRS})
+
+  if (${HDF5_VERSION} GREATER 1.8)
+    message(WARNING "HDF5 will be used such that it is compatible with version 1.8.")
```

Comment:
Isn't the check for version 1.8 already part of the above `find_package(HDF5 ..)` line?

Reply:
Indeed it is. This is because HDF5 versions greater than 1.8 have some api changes in them that are not compatible with 1.8. They can be set to work in "1.8 mode", which is the default behaviour for now, but might need to be changed in the future.